### DISCUSSION TITLE
Unable to build

### DISCUSSION THREAD

**Discussion by BlackPixxel** [iNavFlight/inav#6453] [Q&A] [answered]
Hi!

I would like to build inav, but I cannot get it to work.

I use the WSL approach: https://github.com/iNavFlight/inav/blob/master/docs/development/Building%20in%20Windows%2010%20with%20Linux%20Subsystem.md

When I want to build, I am getting the error from the troubleshooting (Syntax error: "(" unexpected).

I tried WSL1 and WSL2 and did all of the proposed solution steps under troubleshooting, but the error stays the same.

Is there another way to solve this issue?

**Comment by DzikuVx** [comment, marked as answer]
The issue is caused by Windows path being availabe inside WSL. Are ytou sure you follow all the possible solutions? Including a restart of Windows?

**Reply by goodnerdboy** [reply]
Hi
After fixing the registry you have to restart your computer then go back to the cmake step.
If you do not run the cmake again it will keep giving you this error

**Reply by goodnerdboy** [reply]
(As written in the wiki but I personally missed that step at first)

**Reply by BlackPixxel** [reply]
Yes, this is exactly what happened to me. I missed doing that 'cmake ..' again.

After doing that, it works as expected.

Thanks alot for the help guys!

**Comment by NameOfTheDragon** [comment]
Perhaps an easier approach is to install Docker Desktop and enable the WSL2 integration mode. Then you don't have to do any installation or configure anything in your WSL instance to build iNav. Just log into your Ubuntu terminal (I use Windows Terminal), clone the iNav repo, then issue the `sh ./build.sh TARGET` command. The build script creates the docker container with all the necessary tools installed, starts it and runs the build. Docker Desktop enables docker containers to run inside the WSL2 process. Very nice.
